wiki.ros.org/metapackage_status.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+
+"""
+Generates wiki formatted output, which summarizes the status of a release's variants
+using the previous release's distribution cache and accounting for changes.
+
+To use it, checkout the <distro>-devel branch of the metapackages repo at
+
+https://github.com/ros/metapackages
+
+And then update the `path` variable near the top of the script to point to the directory
+containing the metapackages repo and run this script.
+"""
+
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from catkin_pkg.packages import find_packages
+from catkin_pkg.package import InvalidPackage
+from catkin_pkg.package import parse_package_string
+
+from rosdistro import get_cached_distribution
+from rosdistro import get_distribution_file
+from rosdistro import get_index
+from rosdistro import get_index_url
+from rosdistro.dependency_walker import DependencyWalker
+import argparse
+
+parser = argparse.ArgumentParser(
+    description='Generate wiki formatted output for the Maintenance status page.')
+parser.add_argument(
+    'current_distro', metavar='current_distro', type=str, nargs='?',
+    help='The distribution to generate a Wiki page for.', default='kinetic')
+parser.add_argument(
+    'prev_distro', metavar='prev_distro', type=str, nargs='?',
+    help='The distribution to generate a Wiki page for.', default='indigo')
+parser.add_argument(
+  'path', metavar='path', type=str, nargs='?',
+  help='The path to the ros/metapackages Git repo.', default='/tmp/mp_ws/src')
+args = parser.parse_args()
+cur_dist_key = args.current_distro
+prev_dist_key = args.prev_distro
+
+path = args.path
+
+metapackages = dict([(p.name, p) for pth, p in find_packages(path).items()])
+
+keys = [
+    'ros_core',
+    'ros_base',
+    'robot',
+    'viz',
+    'desktop',
+    'perception',
+    'simulators',
+    'desktop_full',
+]
+
+# Get packages which make up each layer of the veriants
+mp_sets = {}
+index = get_index(get_index_url())
+prev_dist = get_cached_distribution(index, prev_dist_key)
+cur_dist = get_cached_distribution(index, cur_dist_key)
+dist_file = get_distribution_file(index, prev_dist_key)
+cur_dist_file = get_distribution_file(index, cur_dist_key)
+dw = DependencyWalker(prev_distro)
```

Comment:
can you tell I haven't run it yet (because Kinetic hasn't yet been added to rosdistro)?